middleware/authMiddleware.js

```diff
@@ -0,0 +1,21 @@
+const jwt = require("jsonwebtoken");
+
+const authenticateToken = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  const token = authHeader && authHeader.split(" ")[1];
+
+  if (token == null) {
+    return res.sendStatus(401);
+  }
+
+  jwt.verify(token, process.env.TOKEN_SECRET, (err, user) => {
+    if (err) {
+      return res.sendStatus(403);
+    }
+
+    req.user = user;
```

Comment:
Zanim to zrobisz to trzeba jeszcze pobrać dane użytkownika z bazy danych. I jeśli je znajdziesz to sprawdzić czy token jest ten sam. I jeśli wszystko jest ok to przypisać user do req.

Reply:
Teraz powinno być ok.
Niestety źle zrozumiałem funkcję pliku contacts.json i mogłem trochę przekombinować